sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -1874,36 +1878,60 @@ export class ApplicationService extends RecordDataModelService<Application> {
   /**
    * Get previous application versions for an application.
    * @param applicationId application id.
-   * @param options method options.
-   * - `limit` number of previous application versions to be returned.
-   * - `loadDynamicData` optionally loads the dynamic data.
    * @returns previous application versions.
    */
   async getPreviousApplicationVersions(
     applicationId: number,
-    options?: { loadDynamicData?: boolean; limit?: number },
   ): Promise<Application[]> {
-    const application = await this.repo.findOne({
-      select: { applicationNumber: true, submittedDate: true },
-      where: { id: applicationId },
-    });
     return this.repo.find({
       select: {
         id: true,
         submittedDate: true,
-        data: !!options?.loadDynamicData as unknown,
       },
       where: {
-        submittedDate: LessThan(application.submittedDate),
-        applicationNumber: application.applicationNumber,
+        parentApplication: { id: applicationId },
+        applicationStatus: ApplicationStatus.Overwritten,
       },
       order: {
         submittedDate: "DESC",
       },
-      take: options?.limit,
     });
   }
 
+  /**
+   * Get last application version.
+   * @param applicationId application id.
+   * @returns last application version.
+   */
+  async getLastApplicationVersion(applicationId: number): Promise<Application> {
```

Comment:
Method name does not align with the logic. Please update the name to something like `getApplicationData`